sklearn/linear_model/_base.py

```diff
@@ -472,11 +480,14 @@ class LinearRegression(MultiOutputMixin, RegressorMixin, LinearModel):
     """
     @_deprecate_positional_args
     def __init__(self, *, fit_intercept=True, normalize=False, copy_X=True,
-                 n_jobs=None):
+                 n_jobs=None, solver="lstsq"):
         self.fit_intercept = fit_intercept
         self.normalize = normalize
         self.copy_X = copy_X
         self.n_jobs = n_jobs
+        if solver not in ["lstsq", "cholesky"]:
+            raise ValueError("Solver must be either `lstsq` or `cholesky`")
```

Comment:
By convention we validate hyperparameters in `fit`.

Reply:
Furthermore it would be great to also expand the error message to report the observed invalid value of the `solver` parameter.

```python
    raise ValueError(f"Solver must be either `lstsq` or `cholesky`, got: {solver}")
```